client/python/datajunction/client.py

```diff
@@ -547,17 +603,24 @@ def sql(  # pylint: disable=too-many-arguments
         return response.json()
 
     @staticmethod
-    def results_to_dataframe(results) -> pd.DataFrame:
+    def process_results(results) -> "pd.DataFrame":
         """
-        Return a pandas dataframe of the results.
+        Return a pandas dataframe of the results if pandas is installed
         """
         if "results" in results and results["results"]:
             columns = results["results"][0]["columns"]
             rows = results["results"][0]["rows"]
-            return pd.DataFrame(
-                rows,
-                columns=[col["name"] for col in columns],
-            )
+            try:
+                return pd.DataFrame(
```

Comment:
@shangyian I added back the import error catch logic you had and then did a try/except here. Pandas is an optional dependency now so as you described, it will work either if someone already has pandas in their environment or if they installed the dj client using `pip install datajunction[pandas]`